webapp/lib/controller.dart

```diff
@@ -267,48 +277,86 @@ bool isProjectStale(model.NeedsReplyData projectData) {
   }
 }
 
-void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
-  projectTimers[selectedProject]?.cancel();
+String getWatchdogTimer(Object data) {
```

Comment:
nit: it's not returning the timer, just the key in the map. Consider renaming to:
```suggestion
String getWatchdogTimerKey(Object data) {
```

Reply:
Renamed it.